test: `:file {name}`

justinmk opened the pull request:
covers #6487

erw7 committed:
win: mch_open_rw: specify S_IWRITE #6487

On Windows, `mch_open_rw` is not actually doing what it claims. This
manifests as "E301: Oops, lost the swap file !!!" when filename is
changed with :file {name}.

Steps to reproduce (covered by test/functional/ex_cmds/file_spec.lua):
    nvim -u NONE
    :edit test
    :file test2
    E301 Oops, lost the swap file!!!

From libuv/src/win/fs.c:
    void fs__open(uv_fs_t* req) {
      ...
      attributes |= FILE_ATTRIBUTE_NORMAL;
      if (flags & _O_CREAT) {
        if (!((req->fs.info.mode & ~current_umask) & _S_IWRITE)) {
          attributes |= FILE_ATTRIBUTE_READONLY;
        }
      }

justinmk committed:
test: `:file {name}`

justinmk commented on src/nvim/macros.h:
```diff
@@ -120,8 +120,10 @@
 #if defined(UNIX)  /* open in rw------- mode */
 # define mch_open_rw(n, f)      os_open((n), (f), (mode_t)0600)
+#elif defined(WIN32)
+# define mch_open_rw(n, f)      os_open((n), (f), S_IREAD | S_IWRITE)
```
@erw7 I included S_IREAD here, if only to be explicit. That matches the `0600` permissions for the UNIX case, and it also matches what Vim does.

erw7 replied:
I think there is no problem.
I closed #6487, please merge this.

justinmk committed:
test/rmdir(): Remove `readonly` attr on Windows.

justinmk committed:
win/test: Enable recover_spec.lua

justinmk committed:
test: helpers.execute() => helpers.feed_command()

justinmk committed:
ci/appveyor: fix cache pattern

justinmk committed:
test/rmdir(): fallback to Nvim delete()

Lua has too many pitfalls here:
- os.execute() requires shell-escaping
- os.execute() has breaking changes between Lua 5.1 and 5.2
- No native way in Lua to handle "readonly" etc. on Windows
